airflow/www/security.py

```diff
@@ -485,12 +485,14 @@ def has_access(self, action_name, resource_name, user=None) -> bool:
 
         has_access = self._has_access(user, action_name, resource_name)
         # FAB built-in view access method. Won't work for AllDag access.
-
         if self.is_dag_resource(resource_name):
+            root_dag_resource_name = resource_name.split(".")[0]
```

Comment:
Should this logic instead go in `can_read_dag` and  `can_edit_dag`?

Reply:
I updated here: https://github.com/apache/airflow/pull/18160/commits/14682d89e4897fc2572a549a0b0e02e341cfb3dc -- do let me know if you don't agree

Reply:
@kaxil Yeah you're correct, this is where it should have gone.

Can a valid dag name include a period? If so, you'll need to test for both the complete version and the root version.